src/query/frontend/semantic/symbol_table.hpp

```diff
@@ -51,6 +51,7 @@ class SymbolTable final {
   const Symbol &at(const Identifier &ident) const { return table_.at(ident.symbol_pos_); }
   const Symbol &at(const NamedExpression &nexpr) const { return table_.at(nexpr.symbol_pos_); }
   const Symbol &at(const Aggregation &aggr) const { return table_.at(aggr.symbol_pos_); }
+  const Symbol &at(const Exists &exists) const { return table_.at(exists.symbol_pos_); }
```

Comment:
Can we make Exists as NamedExpression? As I understood, NamedExpression stores value on Frame as Exists does.

I am pretty confident that we can, I am not sure as what was reasoning to do it as new class. Because if we now that filter is of type Pattern, than expression in filter is used as NamedExpression and functionality should be the same, right?

```
std::unique_ptr<LogicalOperator> MakeExistsFilter(const NamedExpression &exists, AstStorage &storage,
                                                    const SymbolTable &symbol_table,
                                                    const std::unordered_set<Symbol> &bound_symbols) {
```

I think that should be possible.

Reply:
I would not put it under named expression, since indeed it does not have a name and the expression has totally different logic of execution. NamedExpressions are used in returns. This one executes a sub-branch

Reply:
As we talked about this and the problem with using `NamedExpression` is that it would require changing of logic inside its visitor and has different logic, I will resolve this comment